documentation/src/main/asciidoc/user_guide/integrations.adoc

```diff
@@ -11,6 +11,212 @@ Also check the link:$$https://github.com/infinispan/infinispan-spark/tree/master
+import org.infinispan.lucene.directory.DirectoryBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+
+// Create caches that will store the index. Here the Infinispan programmatic configuration is used
+DefaultCacheManager defaultCacheManager = new DefaultCacheManager();
+Cache metadataCache = defaultCacheManager.getCache("metadataCache");
+Cache dataCache = defaultCacheManager.getCache("dataCache");
+Cache lockCache = defaultCacheManager.getCache("lockCache");
+
+// Create the directory
+Directory directory = DirectoryBuilder.newDirectoryInstance(metadataCache, dataCache, lockCache, indexName).create();
+
+// Use the directory in Lucene
+IndexWriterConfig indexWriterConfig = new IndexWriterConfig(new StandardAnalyzer()).setOpenMode(IndexWriterConfig.OpenMode.CREATE_OR_APPEND);
+
+IndexWriter indexWriter = new IndexWriter(directory, indexWriterConfig);
+
+// Index a single document
+Document doc = new Document();
+doc.add(new StringField("field", "value", Field.Store.NO));
+indexWriter.addDocument(doc);
+indexWriter.close();
+
+// Querying the inserted document
+DirectoryReader directoryReader = DirectoryReader.open(directory);
+IndexSearcher searcher = new IndexSearcher(directoryReader);
+TermQuery query = new TermQuery(new Term("field", "value"));
+TopDocs topDocs = searcher.search(query, 10);
+System.out.println(topDocs.totalHits);
+
+----
+
+The _indexName_ in the _DirectoryBuilder_ is a unique key to identify your index. It takes the same role as the path did on filesystem based indexes: you can create several different indexes giving them different names. When you use the same _indexName_ in another instance connected to the same network (or instantiated on the same machine, useful for testing) they will join, form a cluster and share all content. Using a different _indexName_ allows you to store different indexes in the same set of Caches.
+
+The _metadataCache_, _dataCache_ and _lockCache_ are the caches that will store the indexes. More details provided below.
+
+New nodes can be added or removed dynamically, making the service administration very easy and also suited for cloud environments: it's simple to react to load spikes, as adding more memory and CPU power to the search system is done by just starting more nodes.
+
+
+==== Configuration
+Infinispan can be configured as LOCAL clustering mode, in which case it will disable clustering features and serve as a cache for the index, or any clustering mode. A transaction manager is not mandatory, but when enabled the changes to the index can participate in transactions.
+
+Batching was required in previous versions, it's not strictly needed anymore.
+
+As pointed out in the javadocs of link:$$https://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/lucene/directory/DirectotyBuilder.html$$[DirectoryBuilder], it's possible for it to use more than a single cache, using specific configurations for different purposes. Each cache is explained below:
```

Comment:
Update, fixed the link and a few other places where the attribute was not being rendered properly.